### ISSUE TITLE
Can't build Agent on the Raspberry Pi

### ISSUE THREAD

**Issue by mondalaci** [UltimateHackingKeyboard/agent#400]
We'll use Raspberry Pi 3 Model B  computers as the workstations for our assembly stations in the factory, so it's important to be able to run our whole stack on it, Agent included.

I've made a lot of progress and I think I'm very close but [sass fails to build on the pi](https://gist.github.com/mondalaci/78836e903e14bcb1fd2d2043ba1ddd49) when invoking `npm run build`.

Interestingly, [I can build sass on the Pi](https://gist.github.com/mondalaci/fffd8c188cc8748b34528eabd5ca4095) when invoking `npm install node-sass`.

So the question is how can I build sass while building Agent. It'd be great if the build process could be modified to allow this.

**Comment by ert78gb** [comment]
What is the version of node.js when istalled node-sass success?

**Comment by ert78gb** [comment]
The official answer https://github.com/sass/node-sass/releases/tag/v4.5.3
But you installed well the node-sass and get error only when you used. 
What happening when you create a simple scss file and try to build with node-sass?
`node-sass sample.scss sample.css`

**Comment by mondalaci** [comment]
I also use node 8.3.0 on the pi.

As for the official answer, I know that arm is not officially supported by sass, but apparently it can be built, so we should make it work for Agent, too.

I was able to compile the scss to css with your command without any errors.

**Comment by ert78gb** [comment]
Workaround until we find the solution :
Build the whole code on Linux 
Build electron main on arm
Copy the uhk-agent/dist/renderer from Linux to arm

**Comment by mondalaci** [comment]
First up, a slight correction: Both my desktop machine and the pi runs Linux, so I guess "Linux" should be substituted with "Linux-x86" and "arm" with "Linux-arm".

I did as you suggested, and unfortunately Agent [still fails](https://gist.github.com/mondalaci/7c3f283e4914668a16cc34a5b10a0b3e) on the pi. The electron related error message is weird because `node_modules/electron/dist/electron` works perfectly on the pi.

Any furhter ideas?

**Comment by ert78gb** [comment]
I have to install rpi in virtualbox on my machine and try to find what is the error

**Comment by ert78gb** [comment]
what happening when we not use lerna
run the following commands:
- `npm run build` in uhk-agent folder
- `npm run build:renderer` in uhk-web folder
- `npm start` in uhk-agent folder

the order is important

**Comment by mondalaci** [comment]
`npm run build` trigggered the above sass error regarding unsupported arm architecture.

Then I ran `npm run build:electron:renderer` which procued the same sass error.

Finally, `npm start` said "missing script: start".

Currently, this is not a high priority issue, and your virtualbox approach should work, but please make sure that before implementing this, ask me for the current rpi image. That way, you'll be able to get up to speed much faster.

**Comment by mondalaci** [comment]
I've made Agent work on the Pi.

First up, pull https://github.com/fg2it/phantomjs-on-raspberry and copy `rpi-2-3/wheezy-jessie/v2.1.1/phantomjs` to /usr/bin (although installing the deb would have probably been a more proper way).

Then I followed the usual installation procedure of Agent, but normally sass gives problems on the Pi. To resolve it, after `npm i` one has to `cd packages/uhk-web; npm rebuild node-sass`, and continue with the rest of the commands.
